### PR TITLE
Fix float:fromHexString

### LINKED ISSUE #32220: float:fromHexString throws an error when used
**Description:**
`float:fromHexString`  function throws the following error 
`error: {ballerina}TypeCastError {"message":"incompatible types: 'handle' cannot be cast to '(float|error)'"}
        at ballerina.lang.float.1_0_0:fromHexString(float.bal:367)`

**Steps to reproduce:**
```
import ballerina/io;

public function main() {
    io:println(float:fromHexString("0x1a"));
}
```
It gives the same error with or without the 0x prefix (as used in `int:fromHexString` )

**Affected Versions:**
Ballerina Swan Lake Beta 2
Language specification 2021R1
Update Tool 1.3.5

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #32220

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java [hunk 1]
```diff
@@ -29,6 +35,11 @@
 
     public static Object fromHexString(BString s) {
         // TODO: 6/20/19 Implement this
-        return null;
+        try {
+            return Double.parseDouble(s.getValue());
```
Comment (manuranga) [inline langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java:39]: Is [java's hex float format](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf(java.lang.String)) same as Ballerina spec's format?

If yes please all test cases that covers all [production](https://en.wikipedia.org/wiki/Production_(computer_science)) rules.

File: langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java [hunk 2]
```diff
@@ -29,6 +35,11 @@
 
     public static Object fromHexString(BString s) {
         // TODO: 6/20/19 Implement this
```
Comment (manuranga) [inline langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java:37]: remove this line as well.

File: langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java (hunk 1 above)
Comment (gabilang) [reply langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java:38]: Yes, the Java's [HexFloatingPointLiteral](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf(java.lang.String)) has a similar format to the [HexFloatingPointLiteral](https://ballerina.io/spec/lang/2021R1/#HexIndicator). And I've added some test cases to cover the cases mentioned in the [spec](https://ballerina.io/spec/lang/2021R1/#section_5.2.4)

**Review by warunalakshitha — COMMENTED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**